pkg/archive/archive.go

```diff
@@ -136,10 +138,37 @@ func DetectCompression(source []byte) Compression {
 	return Uncompressed
 }
 
-func xzDecompress(archive io.Reader) (io.ReadCloser, <-chan struct{}, error) {
+func xzDecompress(ctx context.Context, archive io.Reader) (io.ReadCloser, error) {
 	args := []string{"xz", "-d", "-c", "-q"}
 
-	return cmdStream(exec.Command(args[0], args[1:]...), archive)
+	return cmdStream(exec.CommandContext(ctx, args[0], args[1:]...), archive)
+}
+
+func gzDecompress(ctx context.Context, buf io.Reader) (io.ReadCloser, error) {
+	disablePigzEnv := os.Getenv("MOBY_DISABLE_PIGZ")
```

Comment:
Can we have a documentation that lists up all supported environment variables?
Also, probably we could add them to `dockerd --help` (in a separate PR).

Reply:
I can add the documentation, RE: environment variables in a separate PR? Does that work?